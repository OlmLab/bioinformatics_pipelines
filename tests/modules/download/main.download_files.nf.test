nextflow_process {

    name "Test Process download_files"
    script "modules/download/main.nf"
    process "download_files"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = "https://example-files.online-convert.com/document/txt/example.txt"
                input[1] = "download_test"
                """
            }
        }

        then {
            assert process.success
            assert process.out.downloaded_files.size() == 1
            assert snapshot(process.out).match()
        }

    }

}
