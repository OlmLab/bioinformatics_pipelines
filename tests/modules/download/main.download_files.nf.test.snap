{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "example.txt:md5,51ce4dcef1fa486499b8b50dcfd91c33"
                ],
                "downloaded_files": [
                    "example.txt:md5,51ce4dcef1fa486499b8b50dcfd91c33"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-30T11:42:45.803604"
    }
}